public/docs/ts/latest/guide/index.jade

```diff
@@ -1,10 +1,501 @@
+
+  Angular doesn't *enforce* these principles. 
+  It won't complain if write a "kitchen sink" component with 3000 lines.
+  
+  Angular does help us *follow* these principles ... by making it easy to factor our
+  application logic into services and make those services available to components through *dependency injection*.
+
+.l-main-section
+<a id="dependency-injection"></a>
+:marked
+  ## Dependency Injection
+figure
+  img(src="/resources/images/devguide/intro/dependency-injection.png" alt="Service" style="float:left; width:200px; margin-left:-40px;margin-right:10px" )
+:marked
+  "Dependency Injection" is a way to supply a new instance of a class 
+  with the fully-formed dependencies it requires. Most dependencies are services.
+  Angular uses dependency injection to provide new components with the services they need.
+<br clear="all">
+:marked
+  Angular can tell which services a component needs by looking at the types of its constructor parameters.
+  We see in this next example that our `HeroListComponent` needs the `HeroService`:
++makeExample('intro/ts/app/hero-list.component.ts', 'ctor')
+:marked
+  When Angular creates a component, it first asks an **Injector** for
+  the services that the component requires. 
+  
+  An `Injector` maintains a container of service instances that it has previously created.
+  If a requested service is not in the container, the injector makes one and adds it to the container
+  before returning the service to Angular. 
+  When all requested servics have been resolved and returned, 
+  Angular can call the component's constructor with those services as arguments.
+  This is what we mean by *dependency injection*.
+  
+  The process of `HeroService` injection looks a bit like this:
+figure
+  img(src="/resources/images/devguide/intro/injector-injects.png" alt="Service" )
+:marked
+  If the `Injector` doesn't have a `HeroService`, how does it know how to make one?
+  
+  In brief, we must have previously registered a **provider** of the `HeroService` with the `Injector`.
+  A provider is something that can create or return a service, typically  the service class itself.
+  
+  We can register providers at any level of the application component tree.
+  We often do so at the root when we bootstrap the application so that
+  the same instance of a service is available everywhere.
++makeExample('intro/ts/app/boot.ts', 'bootstrap')
+:marked
+  Alternatively, we might register at a component level ...
++makeExample('intro/ts/app/hero-list.component.ts', 'providers')
+:marked
+  ... in which case we get a new instance of the
+  service with each new instance of that component.
+  
+  We've vastly over-simplified dependency injection for this overview.
+  We can learn the full story in the [Dependency Injection](dependency-injection.html) chapter.
+  
+  The points to remember are:
+  * dependency injection is wired into the framework and used everywhere.<br><br>
```

Comment:
I'd also mention that it can be used as a standalone system and integrates with other frameworks. One nice thing about the new DI is that it *is* usable standalone, without being wired into the framework.